pkg/api/api.go

```diff
@@ -205,15 +211,15 @@ func (api *API) getProofHandler(r *http.Request) (interface{}, error) {
 }
 
 func (api *API) addHandler(r *http.Request) (interface{}, error) {
-	defer r.Body.Close()
-
-	var byteEntry bytes.Buffer
-	tee := io.TeeReader(r.Body, &byteEntry)
+	b, err := ioutil.ReadAll(r.Body)
```

Comment:
In this approach, the entire request body will be read into memory immediately, whereas the current code only does this *iff* the leaf content is successfully parsed and checked that it is not already in the tLog. 

While the major reason to do this is to defer parsing the `Data` element of the incoming JSON doc (until we absolutely need to, since it might be huge), it also provides some OOM protection from clients sending huge malformed JSON documents to the server. 

If we're still planning to drop the `Data` element then admittedly the current code doesn't provide as much benefit, but in general I'm suspect of using `ioutil.ReadAll()` in a case where the input stream isn't capped in some way.